Add bearingless induction motor analyzers and examples

Anvar-Khamitov opened the pull request:
This PR includes:
- FEMM tool,
- "CrossSectInnerRotorRoundSlots" class,
- Example of using "CrossSectInnerRotorRoundSlots" with FEMM and JMAG tools. For this, run file "example_inner_rotor_round_slots.py" in eMach\examples\mach_cad_examples

elsevers reviewed (changes requested):
There seem to be a lot of "white-space only" changes in here. Are these intentional? 

If they are not, let's back out the changes. It is not a good practice to commit white space changes were inadvertently created (okay to commit them if you are doing this for a purpose, such as cleaning up a file)

Bharat-Ramadas commented:
The added whitespaces are intended. These changes make the files more PEP compliant (mainly [PEP 8](https://pep8.org/)) in terms of spacing and other style guidelines.

elsevers commented:
Can you please add a description to this PR? What functionality are you adding?

Anvar-Khamitov committed:
Add FEMM tool

Anvar-Khamitov committed:
Add class and example for inner rotor round slots

Anvar-Khamitov committed:
Merge branch 'develop' of https://github.com/Severson-Group/eMach into develop

Anvar-Khamitov committed:
Fix material assignment in FEMM tool

Anvar-Khamitov committed:
Add a class and example for inner rotor drop slots

Anvar-Khamitov commented:
@elsevers this PR is ready for review. It includes:
* FEMM tool,
* Cross-section classes "CrossSectInnerRotorRoundSlots" and "CrossSectInnerRotorDropSlots",
* Examples of drawing inner rotor round and drop slots ("example_inner_rotor_round_slots.py" and "example_inner_rotor_drop_slots.py" in eMach\examples\mach_cad_examples).

Anvar-Khamitov committed:
Add cross-section classes and .svg files for round and drop bars

Anvar-Khamitov committed:
Add cross-section classes for stator's right and left slots

Anvar-Khamitov committed:
Fix the links and set_block_prop method in FEMM tool

Anvar-Khamitov committed:
Include round and drop bar components in IM rotor drawing examples

Anvar-Khamitov committed:
Add an example that draws induction motor in FEMM and JMAG

Anvar-Khamitov committed:
Merge commit 'edfac83e2f8104c5fc3796da08632db22c54490f' into develop

Anvar-Khamitov commented:
@elsevers I have updated this PR. It is ready for review. The major updates are:
* Added cross-section classes and .svg files for round and drop bars, and stator's right and left slots.
* Modified examples "example_inner_rotor_round_slots.py" and "example_inner_rotor_drop_slots.py" to draw the rotor core as well as the rotor bars in FEMM and JMAG.
* Added a new example "example_induction_motor.py" that draws induction motor in FEMM and JMAG, including rotor bars and stator windings with two layers.

Anvar-Khamitov committed:
Add new methods and make "set_current" work with time harmonic solver

Anvar-Khamitov committed:
Add partial models for rotor and stator cross-sections

Anvar-Khamitov committed:
Add copy rotate feature to JMAG tool

Anvar-Khamitov committed:
Modify example induction motor not to overwrite an existing file

Anvar-Khamitov committed:
Add classes for BIM rotor iron, rotor bar, and MP winding

Anvar-Khamitov committed:
Add .gitignore to ignore JMAG and FEMM files

Anvar-Khamitov committed:
Add time-harmonic analyzer

Anvar-Khamitov committed:
Add transient analyzer with two time-step sections

Anvar-Khamitov committed:
Add two examples to test BIM evaluator

Anvar-Khamitov committed:
Add classes for BIM machine and BIM operating point

Anvar-Khamitov committed:
Make commit before switching branches

Anvar-Khamitov committed:
Modify force vector analyzer

Anvar-Khamitov committed:
Merge branch 'force_torque_analyzers' into develop

Anvar-Khamitov committed:
Add stator phase winding resistance analyzer

Anvar-Khamitov committed:
Merge branch 'stator_phase_wdg_res_analyzer' into develop

Anvar-Khamitov committed:
Modify torque analyzer to properly calculate torque ripple

Anvar-Khamitov committed:
Merge branch 'force_torque_analyzers' into develop

Anvar-Khamitov committed:
Add scripts for BIM optimization

Anvar-Khamitov committed:
Add additional analyzer configurations

Anvar-Khamitov committed:
Make modifications to BIM Machine

Anvar-Khamitov committed:
Add scaling calculations in T2TSS post-analyzer

Anvar-Khamitov committed:
Fix issues with BIM optimization scripts

Anvar-Khamitov committed:
Add documentation for torque data analyzer

Anvar-Khamitov committed:
Add documentation for force data analyzer

Anvar-Khamitov committed:
Add documentation for stator winding resistance analyzer

Anvar-Khamitov committed:
Merge branch 'force_torque_analyzers' into develop

Anvar-Khamitov committed:
Remove new lines from index file to avoid merge conflicts

Anvar-Khamitov committed:
Remove new line from index file to avoid merge conflicts

Anvar-Khamitov committed:
Merge branch 'stator_phase_wdg_res_analyzer' into develop

Anvar-Khamitov committed:
Merge branch 'force_torque_analyzers' into develop

Anvar-Khamitov committed:
Make misc changes to BIM optimization and analyzer scripts

Anvar-Khamitov committed:
Add scripts for BIM double cage and FEMM boundary conditions

Anvar-Khamitov committed:
Fix the issue with 'PYIDispatch' object (and clean the code)
